relay-monitors/src/lib.rs

```diff
@@ -176,7 +193,20 @@ pub fn process_check_in(payload: &[u8]) -> Result<Vec<u8>, ProcessCheckInError>
         return Err(ProcessCheckInError::InvalidEnvironment);
     }
 
-    Ok(serde_json::to_vec(&check_in)?)
+    static NAMESPACE: OnceCell<Uuid> = OnceCell::new();
+    let namespace = NAMESPACE
+        .get_or_init(|| Uuid::new_v5(&Uuid::NAMESPACE_URL, b"https://sentry.io/crons/#did"));
+
+    // Use the monitor slug as the routing key hint. This helps ensure monitor
+    // check-ins are processed in order by routing check-ins from the same monitor
+    // to the same place.
+    let project_id_slug_key = format!("{}-{}", project_id, check_in.monitor_slug);
+    let routing_hint = Uuid::new_v5(namespace, project_id_slug_key.as_bytes());
```

Comment:
There's probably a more efficient way to slap these together, let me see if I can figure it out